pkg/codec/byte_string_modifier.go

```diff
@@ -1,12 +1,79 @@
 package codec
 
 import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
 	"fmt"
 	"reflect"
 
 	"github.com/smartcontractkit/chainlink-common/pkg/types"
 )
 
+type ExampleAddressModifier struct{}
+
+func (m *ExampleAddressModifier) EncodeAddress(bts []byte) (string, error) {
+	if len(bts) > 32 {
+		return "", errors.New("upexpected address byte length")
+	}
+
+	normalized := make([]byte, 32)
+
+	// apply byts as big endian
+	copy(normalized[:], bts[:])
+
+	return base64.StdEncoding.EncodeToString(normalized), nil
+}
+
+func (m *ExampleAddressModifier) DecodeAddress(str string) ([]byte, error) {
+	decodedBytes, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(decodedBytes) != 32 {
+		return nil, errors.New("unexpected address byte length")
+	}
+
+	return decodedBytes, nil
+}
+
+func (m *ExampleAddressModifier) Length() int {
+	return 32
+}
+
+func ExampleAddressBytesToStringModifier() {
+	type onChainNested struct {
+		X []byte
+	}
+
+	type onChain struct {
+		A [32]byte
+		B onChainNested
+	}
+
+	encoder := &ExampleAddressModifier{}
+	mod := NewPathTraverseAddressBytesToStringModifier([]string{"B.X"}, encoder, true)
+
+	// call RetypeToOffChain first with empty itemType to set base types
+	offChainType, _ := mod.RetypeToOffChain(reflect.TypeOf(&onChain{}), "")
+
+	fmt.Println("offChainType:")
```

Comment:
This stays?

Reply:
This is part of the example and should stay, yes.